lib/src/disposable/transformer/transformations.dart

```diff
@@ -169,10 +169,17 @@ void transformLibraryToImportDartAsync(
       directive.uri.stringValue == 'dart:async');
   if (hasDartAsyncImport) return;
 
+  // If this file is a part (meaning the imports are defined in a separate,
+  // parent file), then nothing can be done.
+  final isPartOf =
+      unit.directives.any((directive) => directive is PartOfDirective);
```

Comment:
It will only be an issue if `dart:async` isn't imported in the parent file, which I think will be pretty rare given that these are files that are already dealing with disposable APIs. If that does happen, there will be a runtime error.. I suppose there should at least be some documentation around this